### PR TITLE
Add toggle for sending status

### PR DESCRIPTION
**Pull request by gburning**

### CODE REVIEW

File: extension/demo/src/inject/inject.js [hunk 1]
```diff
@@ -29,9 +29,13 @@ document.addEventListener('flattr-subscription', event => {
   sendMessage('subscription', { type: 'subscription', subscription })
 })
 
+// TODO: Should we send something else if there is no payload?
+// TODO: Should we send something else if there emitStatus is not true?
+// TODO: Move get-emit-status to request-payload?
```
Comment (gburning) [inline extension/demo/src/inject/inject.js:34]: Should be addressed

File: extension/demo/src/modules/constants.js [hunk 2]
```diff
@@ -2,12 +2,14 @@
...
-export const TTL = 'expiresAt'
 const apiDomain = env === 'production' ? 'flattr.com' : 'flattr.test'
 const apiProtocol = env === 'production' ? 'https' : 'http'
 export const API_BASE = `${apiProtocol}://api.${apiDomain}`
 export const API_BASE_WEB = `${apiProtocol}://${apiDomain}`
+
+export const STORAGE_KEY_ACCESS_TOKEN = 'accessToken'
+export const STORAGE_KEY_SUBSCRIPTION = 'subscription'
+export const STORAGE_KEY_EMIT_STATUS = 'emitStatus'
+export const STORAGE_KEY_IS_PAYING = 'isPaying' // TODO: Not in use. Remove?
```
Comment (gburning) [inline extension/demo/src/modules/constants.js:13]: Remove?

File: extension/demo/src/normalize.css [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+  display: none;
+}
+
+/**
+ * Add the correct display in IE 10.
+ */
+
+[hidden] {
+  display: none;
+}
```
Comment (Karagath) [inline extension/demo/src/normalize.css:349]:
```suggestion
}

```

File: extension/demo/src/bg/background.js [hunk 4]
```diff
@@ -24,62 +31,89 @@ function onPopupOpenApps () {
   })
 }
 
-async function onToken (data) {
-  const { accessToken } = data
+function onSetEmitStatus (data) {
```
Comment (Karagath) [inline extension/demo/src/bg/background.js:34]: What does emit status mean?

File: extension/demo/src/bg/background.js [hunk 5]
```diff
@@ -24,62 +31,89 @@ function onPopupOpenApps () {
...
   return { isAuthenticated }
 }
 
-async function requestPayload () {
-  const response = {}
-  response[PAYLOAD] = await storage.get(PAYLOAD)
+function onSetSubscription (subscription) {
+  return browser.storage.local.set({
+    [STORAGE_KEY_SUBSCRIPTION]: subscription
+  })
```
Comment (Karagath) [inline extension/demo/src/bg/background.js:61]:
```suggestion
  return storage.set({ [STORAGE_KEY_SUBSCRIPTION]: subscription })
```

File: extension/demo/src/modules/constants.js (hunk 2 above)
Comment (Karagath) [inline extension/demo/src/modules/constants.js:13]:
```suggestion
```

File: extension/demo/src/bg/background.js (hunk 4 above)
Comment (gburning) [reply extension/demo/src/bg/background.js:34]: Yeah, the name is a little tricky. But it controls whether the payload is sent or not.

Ideas for other names are welcome.

`emitPayload`? `sendPayload`? `shouldSendPayload`?

File: extension/demo/src/bg/background.js [hunk 6]
```diff
@@ -24,62 +31,99 @@ function onPopupOpenApps () {
...
+  }
+
+  return {
+    isAuthenticated,
+    sendPayload
+  }
+}
 
-  return response
+// TODO: Should we send something else if sendPayload is not true?
```
Comment (gburning) [inline extension/demo/src/bg/background.js:83]: ?

File: extension/demo/src/inject/inject.js [hunk 7]
```diff
@@ -29,8 +29,9 @@ document.addEventListener('flattr-subscription', event => {
   sendMessage('subscription', { type: 'subscription', subscription })
 })
 
+// TODO: Should we send something else if there is no payload?
```
Comment (gburning) [inline extension/demo/src/inject/inject.js:32]: ?

File: publisher-website/demo/main.js [hunk 8]
```diff
@@ -66,7 +66,7 @@ let tries = 0
 // TODO: Request a real payload from the extension and only verify if it is returned
```
Comment (gburning) [inline publisher-website/demo/main.js:66]:
```suggestion
```

File: extension/demo/src/bg/background.js (hunk 6 above)
Comment (leihog) [reply extension/demo/src/bg/background.js:83]: No I don't think so.

**Review by Flattr-Henke — APPROVED** [review]

**Pull request merged**